modules/table/src/main/java/org/apache/ignite/internal/table/distributed/schema/SchemaVersions.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.schema;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+
+/**
+ * Used to work with schema versions as seen at a timestamp (applying logic of Schema Synchronization).
+ */
+public interface SchemaVersions {
+    /**
+     * Obtains a schema version at a timestamp in obedience to rules of Schema Synchronization.
+     *
+     * @param timestamp Timestamp at which to obtain a version.
+     * @param tableId ID of the table which schema to return.
+     * @return Future that will complete with the version.
+     */
+    CompletableFuture<Integer> schemaVersionAt(HybridTimestamp timestamp, int tableId);
+
+    /**
+     * Obtains a schema version at 'now' in obedience to rules of Schema Synchronization.
+     *
+     * @param tableId ID of the table which schema to return.
+     * @return Future that will complete with the version.
+     */
+    CompletableFuture<Integer> schemaVersionAtNow(int tableId);
```

Comment:
Do you think this method is needed? we can always pass `org.apache.ignite.internal.hlc.HybridClock#now` as an argument.

Reply:
This will require us to pass `HybridClock` instance to every KV/record table view. I believe that in this case one extra method to hide this dependency is better than passing the dependency (the clock) around.